### ISSUE TITLE
Memory leak with python 3

### ISSUE THREAD

**Issue by caglorithm** [numba/numba#4219]
Hello everyone. I found a memory leak in numba when using python 3. This does not happen when using python 2. It leaks only with the `for i in range(20)` loop. It leaks only when the value is returned, otherwise no leak. It leaks only when the array is copied with `.copy()` as shown below. 

Here is the reproducer:

```python
import numpy as np
import numba

@numba.njit
def return_tuple(input_array):
    for i in range(20):
        ret = input_array.copy()
    return ret

input_array = np.zeros((1000, 1000))
stats = numba.runtime.rtsys.get_allocation_stats()
last_leak = stats[0]-stats[1]
for i in range(10):
    return_tuple(input_array)
    stats = numba.runtime.rtsys.get_allocation_stats()
    print("leak: {}, total leak: {}".format((stats[0]-stats[1])-last_leak, stats[0]-stats[1]))
    last_leak = stats[0]-stats[1]
```

output
```
leak: 19, total leak: 19
leak: 19, total leak: 38
leak: 19, total leak: 57
leak: 19, total leak: 76
leak: 19, total leak: 95
leak: 19, total leak: 114
leak: 19, total leak: 133
leak: 19, total leak: 152
leak: 19, total leak: 171
leak: 19, total leak: 190
```
Python version 3.7.3, numba version '0.43.1'.

I have no idea what could cause this, the underlying numba code is way over my head. Good luck and thank you a lot for all the great work so far.

**Comment by sklam** [comment]
Thanks for the report.  The issue has been resolved in version 0.44.1.

**Comment by sklam** [comment]
Fixed by #4183
